### ISSUE TITLE
@match *://* does not match anything. but @include does.

### ISSUE THREAD

**Issue by CoolCmd** [greasemonkey/greasemonkey#2223]
FAIL:
```
// ==UserScript==
// @name        1
// @namespace   2
// @match       *://*
// @version     1
// @grant       none
// ==/UserScript==
console.log('CoolCmd');
```

OK:
```
// ==UserScript==
// @name        1
// @namespace   2
// @include       *://*
// @version     1
// @grant       none
// ==/UserScript==
console.log('CoolCmd');
```

In the previous version of GM @match works as expected.

**Comment by CoolCmd** [comment]
FF 39, GM 3.2

**Comment by arantius** [comment]
I could have sworn a literal scheme was always required for `@match`, but `*` seems to be supported in all documentation I can find.

**Comment by cletusc** [comment]
https://developer.chrome.com/extensions/match_patterns

```
<url-pattern> := <scheme>://<host><path>
<scheme> := '*' | 'http' | 'https' | 'file' | 'ftp'
<host> := '*' | '*.' <any char except '/' and '*'>+
<path> := '/' <any chars>
```

The syntax is pretty strict, but `*://*` does seem valid as far back as I remember.

**Comment by CoolCmd** [comment]
cletusc, yes, I forget to paste this link.
* = http & https

**Comment by janekptacijarabaci** [comment]
The easiest solution is probably to use: ```*://*/*```

But... Otherwise it is not entirely clear...
See also http://code.google.com/p/chromium/issues/detail?id=318195

**Comment by arantius** [comment]
Tested briefly with:

```javascript
// ==UserScript==
// @name         Test `@match *://*`
// @match        *://*
// @grant        none
// ==/UserScript==

alert('match *://* is a go!');
```

In tampermonkey. The behavior was confusing, so I continued...

I interpret the docs as linked above to mean `<any chars>` does not include the empty set, i.e. the path must be non-empty.  In the same way that `<any char except '/' and '*'>+` does not include the empty string, just written better/more literally.

Moreover, the chromium source has a [special case](https://chromium.googlesource.com/chromium/src.git/+/77cb0effc7d4c844f07bb380827a7b1a2cfd2c3c/extensions/common/url_pattern.cc#476) for a path of `/*`, and otherwise uses [`::MatchPattern()`](https://chromium.googlesource.com/chromium/src.git/+/d97eede0d9eb25ebffc918c8ebfcf9a9d9703a88/base/strings/pattern.cc#157) in which [the empty pattern matches only the empty string](https://chromium.googlesource.com/chromium/src.git/+/d97eede0d9eb25ebffc918c8ebfcf9a9d9703a88/base/strings/pattern.cc#89).

Aha!  The confusing behavior I observed with Tampermonkey is that `@match *://*` only matched when the path was empty, which is exactly what it's supposed to do (per the code ...).  I just didn't realize what was going on with the set of arbitrary URLs I was using (and the way they often forwarded from an empty to a non-empty path).

**TL;DR** The `@match` pattern `*://*` will *by design* only match a URL with an empty path.  If you want to use `@match` to match every URL (and for some reason avoid `@include *`), the correct syntax is `@match *://*/*` -- the path wildcard is not optional.
